retworkx/visualization/pydot.py

```diff
@@ -0,0 +1,116 @@
+        dictionary of Graphviz edge attributes to be associated with the edge
+        in the visualization file. The key and value of this dictionary
+        must be a string.
+    :param dict graph_attr: An optional dictionary that specifies any Graphviz
+        graph attributes for the visualization. The key and value of this
+        dictionary must be a string.
+    :param str filename: An optional path to write the visualization to. If
+        specified the return type from this function will be ``None`` as the
+        output image is saved to disk.
+    :param str image_type: The image file format to use for the generated
+        visualization. The support image formats are:
+        ``'canon'``, ``'cmap'``, ``'cmapx'``, ``'cmapx_np'``, ``'dia'``,
+        ``'dot'``, ``'fig'``, ``'gd'``, ``'gd2'``, ``'gif'``, ``'hpgl'``,
+        ``'imap'``, ``'imap_np'``, ``'ismap'``, ``'jpe'``, ``'jpeg'``,
+        ``'jpg'``, ``'mif'``, ``'mp'``, ``'pcl'``, ``'pdf'``, ``'pic'``,
+        ``'plain'``, ``'plain-ext'``, ``'png'``, ``'ps'``, ``'ps2'``,
+        ``'svg'``, ``'svgz'``, ``'vml'``, ``'vmlz'``, ``'vrml'``, ``'vtx'``,
+        ``'wbmp'``, ``'xdot'``, ``'xlib'``. It's worth noting that while these
+        formats can all be used for generating image files when the ``filename``
+        kwarg is specified, the Pillow library used for the returned object can
+        not work with all these formats.
+    :param str method: The layout method/Graphviz command method to use for
+        generating the visualization. Available options are ``'dot'``,
+        ``'twopi'``, ``'neato'``, ``'circo'``, ``'fdp'``, and ``'sfdp'``.
+        You can refer to the
+        `Graphviz documentation <https://graphviz.org/documentation/>`__ for
+        more details on the different layout methods. By default ``'dot'`` is
+        used.
+
+    :returns: A ``PIL.Image`` object of the generated visualization, if
+        ``filename`` is not specified. If ``filename`` is specified then
+        ``None`` will be returned as the visualization was written to the
+        path specified in ``filename``
+    :rtype: PIL.Image
+    """
+    if not HAS_PYDOT:
+        raise ImportError(
+            "Pydot and Pillow are necessary to use pydot_draw() "
+            "it can be installed with 'pip install pydot pillow'"
+        )
+
+    dot_str = graph.to_dot(node_attr_fn, edge_attr_fn, graph_attr)
+    dot = pydot.graph_from_dot_data(dot_str)[0]
+    if image_type is None:
+        output_format = "png"
+    else:
+        output_format = image_type
+
+    if method is None:
+        prog = "dot"
+    else:
+        prog = method
+
+    if not filename:
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmp_path = os.path.join(tmpdirname, "dag.png")
+            dot.write(tmp_path, format=output_format, prog=prog)
+            with Image.open(tmp_path) as temp_image:
+                image = temp_image.copy()
```

Comment:
Is this a work around for https://github.com/Qiskit/qiskit-terra/issues/6005 ?

Reply:
Yeah I think so, although I saw the fix first from this issue: https://github.com/Qiskit/qiskit-terra/issues/6256